modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -3265,6 +3265,66 @@ protected V getAndRemove0(final K key) throws IgniteCheckedException {
         });
     }
 
+    /** {@inheritDoc} */
+    @Override public Long ttl(K key) {
+        boolean statsEnabled = ctx.statisticsEnabled();
+
+        long start = statsEnabled ? System.nanoTime() : 0L;
+
+        A.notNull(key, "key");
+
+        Long ttlVal = ttl0(key);
+
+        if (statsEnabled)
+            metrics0().addRemoveAndGetTimeNanos(System.nanoTime() - start);
+
+        return ttlVal;
+    }
+
+    /**
+     *
+     * @param key
+     * @return
+     */
+    protected Long ttl0(final K key) {
+        GridCacheEntryEx entry = entryEx(key);
```

Comment:
This approach does not seem correct to me. It will only work if the current node is an "affinity" node (it means that it hosts a primary or backup partition for the key).

For instance, the following test fails:
```
    @Test
    public void testTtlMethodsFromClientNode() throws Exception {
        factory = CreatedExpiryPolicy.factoryOf(new Duration(TimeUnit.SECONDS, 100_000));

        disableEagerTtl = true;

        startGridsMultiThreaded(gridCount());

        IgniteEx client = startClientGrid(gridCount());

        IgniteCache c = client.cache(DEFAULT_CACHE_NAME);

        c.put(12, 12);
        
        assertEquals("Unexpected value!", c.get(12), 12);
        assertTrue("TTL equals to zero!", c.ttl(12) > 0);
    }
```

I think the implementation should work in the way `IgniteCache#contains(K key)` works.